core/builtins/message/__init__.py

```
@@ -296,21 +304,25 @@ async def wait_confirm(self, message_chain=None, quote=True, delete=True, timeou
         try:
             await asyncio.wait_for(flag.wait(), timeout=timeout)
         except asyncio.TimeoutError:
-            if message_chain and delete:
+            if send and delete:
                 await send.delete()
             raise WaitCancelException
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)